main.go

```diff
@@ -1,84 +1,278 @@
 package main
 
 import (
-	"fmt"
 	"os"
 
 	"github.com/drone/drone-go/drone"
-	"github.com/drone/drone-go/plugin"
-	"github.com/drone/drone-go/template"
+	"github.com/joho/godotenv"
+	"github.com/Sirupsen/logrus"
+	"github.com/urfave/cli"
 )
 
-var (
-	buildCommit     string
-	defaultTemplate = `<strong>{{ uppercasefirst build.status }}</strong> <a href="{{ system.link_url }}/{{ repo.owner }}/{{ repo.name }}/{{ build.number }}">{{ repo.owner }}/{{ repo.name }}#{{ truncate build.commit 8 }}</a> ({{ build.branch }}) by {{ build.author }} in {{ duration build.started_at build.finished_at }} </br> - {{ build.message }}`
-)
+var revision string // build number set at compile-time
 
 func main() {
-	fmt.Printf("Drone HipChat Plugin built from %s\n", buildCommit)
+	app := cli.NewApp()
+	app.Name = "hipchat plugin"
+	app.Usage = "hipchat plugin"
+	app.Action = run
+	app.Version = revision
+	app.Flags = []cli.Flag{
+
+		//
+		// plugin args
+		//
+		cli.StringFlag{
+			Name: "url",
+			Usage: "HipChat server URL",
+			Value: "https://api.hipchat.com",
+			EnvVar: "PLUGIN_URL",
+		},
+		cli.StringFlag{
+			Name: "auth_token",
+			Usage: "HipChat server URL",
+			EnvVar: "HIPCHAT_AUTH_TOKEN,PLUGIN_AUTH_TOKEN",
+		},
+		cli.StringFlag{
+			Name: "room",
+			Usage: "ID or URL encoded name of the room",
+			EnvVar: "PLUGIN_ROOM",
+		},
+		cli.StringFlag{
+			Name: "from",
+			Usage: "A label to be shown",
+			Value: "drone",
+			EnvVar: "PLUGIN_FROM",
+		},
+		cli.BoolFlag{
+			Name: "notify",
+			Usage: "Whether this message should trigger a user notification (change the tab color, play a sound, notify mobile phones, etc). Each recipient's notification preferences are taken into account",
+			EnvVar: "PLUGIN_NOTIFY",
+		},
+		cli.StringFlag{
+			Name: "template",
+			Usage: "A handlebars template to create a custom payload body.",
+			Value: "<strong>{{ uppercasefirst build.status }}</strong> <a href=\"{{ build.link_url }}\">{{ repo.owner }}/{{ repo.name }}#{{ truncate build.commit 8 }}</a> ({{ build.branch }}) by {{ build.author }} in {{ duration build.started_at build.finished_at }} </br> - {{ build.message }}",
```

Comment:
As per the [0.5 docs](http://readme.drone.io/0.5/usage/notifications/) `{{ duration build.started_at build.finished_at }}` will no longer work. It should be `{{ duration build.started build.finished }}`. 

Similarly, instead of `{{ build.link_url }}` it is now `{{ build.link }}`.